Add stable releases

kjarosh opened the pull request:
This PR adds "Stable Releases" section to downloads (if there are any) and replaces the default download links with the latest stable version (or the latest nightly if there's no stable). By default 3 last minor, and 2 last major releases are shown. "Compatibility" still uses nightly for measuring progress.

This PR may be tested by switching the repo in config to `kjarosh/ruffle` (there are some test releases).

kjarosh committed:
Add stable releases

kjarosh committed:
Rename "nightlies" to "releases"

Dinnerbone reviewed (approved):
LGTM

I wonder why it had a random reformat... do we not auto format in this repo? 🤔

kjarosh commented:
Hmmm... Running `prettier . --write` gives me 14 more changed files, so I guess formatting code is only a suggestion in this repo :smile:

danielhjacobs commented:
Fun fact, back when we used Jekyll, we had a small setup for stable releases, but it was removed with the switch to React:

https://github.com/ruffle-rs/ruffle-rs.github.io/blob/085ac0897e6d8f9b9a0cb71618f06432febfc950/index.html#L245-L247

danielhjacobs commented:
That setup was "5 last stable releases", no separation of major/minor.

kjarosh commented:
Fixed searching for stable releases. Currently we have to traverse all releases to find out there's no stable releases (which takes around 15s, 80 releases per page, ~15 pages).

Also changed `maxMajor` to 1 in order not to traverse all releases when we have only one major version.
